### ISSUE TITLE
Allow maya-usd to build against Qt in devkit

### ISSUE THREAD

**Issue by dgovil** [Autodesk/maya-usd#1070]
**Is your feature request related to a problem? Please describe.**
The current build system requires specifying an external Qt location for building features that require Qt. (as described here: https://github.com/Autodesk/maya-usd/issues/1005). Since that issue is closed, I am opening this one to track this specific request. 

This is prohibitive for certain companies where Qt's licensing model and legalities push them towards recommending the commercial license out of caution for dealing with the Qt source. 

It would be great if we could piggy back off of the devkit supplied Qt version to avoid having to deal with an independent licensing system. It would also simplify the build process for people if that could happen.

**Describe the solution you'd like**
Modify the build system to allow working with the devkit supplied Qt libs

Thanks!

**Comment by kxl-adsk** [comment]
Thank you for logging this one @dgovil. This is really a request for Maya and we have already raised it internally in the past. I will keep it open here for now and see if anything moved internally.

**Comment by dgovil** [comment]
This has become a blocker for Maya 2024 unfortunately on arm64.
There's no arm64 version of Qt5 available, and it still isn't possible to build against the Qt libs with Maya.

Does Autodesk have instructions on building Qt from source that they can share so I can match your setup at least?
Otherwise this prevents all UFE use

**Comment by dgovil** [comment]
So since this was a blocker for us adopting Maya 2024, I started reworking the CMake a bit.
Something about the way the Maya devkit Qt is laid out wasn't being conducive to how CMake's FindPackage expected it, so I just duplicated enough of what CMake does to make it work.

Here's a patch for anyone interested. It seems to work fine for Maya 2022 and above.  
[devkit.patch](https://github.com/Autodesk/maya-usd/files/11492158/devkit.patch)

I will leave this issue open though because I do think its something that ideally the project should solve. I don't however know enough details about the Linux and Windows side to submit my changes as a cleaned up patch.
